nikola/plugins/misc/taxonomies_classifier.py

```diff
@@ -240,10 +240,12 @@ def _postprocess_path(self, path, lang, append_index='auto', dest_type='page', p
                 path = ['rss']
             elif len(path) == 0 or append_index == 'always':
                 path = path + [os.path.splitext(self.site.config['INDEX_FILE'])[0]]
+            elif len(path) > 0 and append_index == 'never':
+                path[-1] = os.path.splitext(path[-1])[0]
```

Comment:
Because in the past, part of the code assumed that `'never'` never had the extension there, while other parts assumed that it had. This code assumed it wasn't there so it could simply add it. Now `'never'` means "change extension if necessary", so now it needs `splitext` to get rid of the old one.